### ISSUE TITLE
EarningsPerShareDiluted is not rendered as decimal

### ISSUE THREAD

**Issue by gmconte** [ammasjk/xbrlj#2]
I'm looking at the Earnings per share value, which is not like all the others (its "decimal" value is actually positive), and the value in the resulting JSON doesn't have any decimals.

For example:
* doc value = (4.15)
* json value = -4

I'm trying to understand what could be wrong and I can see that this piece of code could be the problem:

from PresentationSerializer.makeFact()

> if (fact.getLongValue() != null) {
                long value = fact.getLongValue();
                if (node.isNegated()) {
                    value = -value;
                }
                object.addProperty(VALUE, value);
            } else if (fact.getDoubleValue() != null) {
                double value = fact.getDoubleValue();
                if (node.isNegated()) {
                    value = -value;
                }
                object.addProperty(VALUE, value);
            } else {
                object.addProperty(VALUE, fact.getValue());
            }

In the debugger, I can see that the getLongValue return the integer number, while double value is null.

### Possible Solution
I'm not sure of the reason for this specific order in selecting the value, but if that cannot be changed, we could have the longValue skipped in case decimals is a positive value.

What do you think?

**Comment by gmconte** [comment]
I did a search for fact.getLongValue() and there are a few additional places that may require the same kind of change

**Comment by gmconte** [comment]
forgot to paste the file url I used for this debugging:
https://www.sec.gov/ix?doc=/Archives/edgar/data/1318605/000156459020004475/tsla-10k_20191231.htm

**Comment by ammasjk** [comment]
Negative values are usually shown in parenthesis in accounting / finance. So, this is actually a negative value.

**Comment by gmconte** [comment]
Yes, but this particular value (whether positive or negative) has a
positive number of decimal and it's not in thousands or millions, so the
decimal part is actually meaningful.

For example 100 thousands is presented as long value = 100 and decimals = -3
While 4.75 is presented as long value 4 and decimals=2.

The problem I have is that a value of 4.75 is reported in the JSON as 4,
discarding all the decimals par, which I want to keep

I don't understand why the double value is null, but changing the check to
`if(fact.getLongValue() != null and decimals < 0)` will force the code to
return `fact.getValue()` instead, which contains the value I want "4.75".


Hope that clarifies what I mean...

**Comment by ammasjk** [comment]
I did check in a fix before I responded that addresses the issue you raised. It is a valid issue. The code was trying to compensate for errors in some XBRL files, by checking if a double value needed to be treated as a double. However, this check did not consider the absolute difference creating the error.

**Comment by gmconte** [comment]
You're right. that solved my issue.
Thanks!
